### PR TITLE
Fix scheduler example

### PR DESCRIPTION
**Pull request by mjethanandani**
Hi Aseem, I have made these changes on top of your diffs. The easiest way would be for you to accept these changes in your branch and work from there. Note, you will get my changes for issue 125, which you will need to be able even compile this example. Ideally, we should merge 125 before making any more changes. Concentrate on the a.1.5 example file, ignore the rest of the changes.

The example is still not compiling, but most of the "syntax" errors are gone. What we are left with is the logic piece. And here is what needs to be determined next. Either the error is in our module or the example. Basically, it is failing on the when statement:

```
    when "derived-from-or-self(../../traffic-policy:type, " +
         "'qos-types:ipv4-diffserv-policy-type') or " +
         "derived-from-or-self(../../traffic-policy:type, " +
         "'qos-types:ipv6-diffserv-policy-type') or " +
         "derived-from-or-self(../../traffic-policy:type, " +
         "'qos-types:diffserv-policy-type')" {

```

So either we have to further relax the when statement by adding the policy type of `queue-policy-type` and `scheduler-policy-type` or we change the type to one of the above types. I do not know what is the correct answer.

```
libyang err : When condition "derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='SCHEDULING-POLICY'][type='iana-qos-types:scheduler-policy-type']/classifier[name='ALL']/inline/filter[type='iana-qos-types:filter-match-all'][logical-not='false']/ietf-diffserv:match-all".)
libyang err : When condition "derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='DATA']/action[type='iana-qos-types:queue']/ietf-diffserv:queue".)
libyang err : When condition "derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='DATA']/inline/filter[type='iana-qos-types:traffic-group-name'][logical-not='false']/ietf-diffserv:traffic-group".)
libyang err : When condition "derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='VIDEO']/action[type='iana-qos-types:queue']/ietf-diffserv:queue".)
libyang err : When condition "derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='VIDEO']/inline/filter[type='iana-qos-types:traffic-group-name'][logical-not='false']/ietf-diffserv:traffic-group".)
libyang err : When condition "derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='VOICE']/action[type='iana-qos-types:queue']/ietf-diffserv:queue".)
libyang err : When condition "derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv4-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:ipv6-diffserv-policy-type') or derived-from-or-self(../../../traffic-policy:type, 'qos-types:diffserv-policy-type')" not satisfied. (Data location "/ietf-traffic-policy:policies/policy[name='QUEUEING-POLICY'][type='iana-qos-types:queue-policy-type']/classifier[name='VOICE']/inline/filter[type='iana-qos-types:traffic-group-name'][logical-not='false']/ietf-diffserv:traffic-group".)
YANGLINT[E]: Failed to parse input data file "yang/example-qos-configuration-a.1.5.xml".
failed (error code: 0)
```

### CODE REVIEW

**Review by aseemavi — APPROVED** [review]

File: src/yang/example-qos-configuration-a.1.5.xml [hunk 1]
```diff
@@ -19,14 +19,14 @@
           <type>qt:traffic-group-name</type>
           <logical-not>false</logical-not>
           <traffic-group
-    	  xmlns="urn:ietf:params:xml:ns:yang:ietf-queue-policy">
+	      xmlns="urn:ietf:params:xml:ns:yang:ietf-diffserv">
```
[the remark this answers is not in the document]
Comment (aseemavi) [reply src/yang/example-qos-configuration-a.1.5.xml:22]: too many errors in the main code base.. may be you can first commit this diff..

[the remark this answers is not in the document]
Comment (mjethanandani) [reply src/yang/example-qos-configuration-a.1.5.xml:22]: Have committed my other fix to take care of errors. At this point, we are still facing the same set of errors in this example as I explained above.

We should try to close this by getting on a call or something.

[the remark this answers is not in the document]
Comment (aseemavi) [reply src/yang/example-qos-configuration-a.1.5.xml:22]: Hi Mahesh,

Let me try with latest. We can get on a call by noon today if it still doesn't work by then.

**Comment by aseemavi** [comment]
create a new PR. It works!

**Comment by aseemavi** [comment]
closing it and replaced with #128

**Pull request closed without merging**